bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java

```diff
@@ -96,6 +101,22 @@ public void removedService(final ServiceReference reference, final Object servic
             }
         };
         thingRegistryServiceTracker.open();
+        linkRegistryServiceTracker = new ServiceTracker(this.bundleContext, ItemChannelLinkRegistry.class.getName(),
```

Comment:
Okay, another question: Why don't the while ThingHandlers using declarative services?
It is okay to do so, just want to understand why we don't use it.